Homework Week 2

gulyener opened the pull request:
This week I didn't continue with last week's homework because it was Class based and I wanted to use functional components in this one.

ArcoMul commented on homework-submission/hw-todolist/src/components/TodoList/index.jsx:
```diff
@@ -0,0 +1,36 @@
+              <button onClick={() => deleteTodo(todo.id)}>Delete todo</button>
+            </td>
+            <td>
+              <button onClick={() => markTodo(todo.id, todo.done)}>Mark todo</button>
```
Would be cool if the text of the 'Mark todo' button is 'Mark done' at fist and changes to 'Mark todo' once a todo is done

ArcoMul commented on homework-submission/hw-todolist/src/components/TodoList/index.jsx:
```diff
@@ -0,0 +1,36 @@
+          >
+            <td>{todo.description}</td>
+            <td>{todo.deadline}</td>
+            {'      '}
```
Why are these spaces needed?

gulyener replied:
You're right I deleted it:)

gulyener commented:
Thank you for the review @ArcoMul. About the delete function, I couldn't see the bug you mentioned but I changed the id's just in case. I don't know if it's correct now. I also changed TodoList component to TodoTable since it's a table. Should I create Todo Items in a separate component?

ArcoMul commented:
Great improvements! `uuid` did fix the bug. What happened is that some todos ended up having the same id because you used `todos.length`
